### PR TITLE
Index set_controller and set_bond calls for controller list

### PR DESCRIPTION
**Pull request by Nexus2k**
https://github.com/w3f/SecOps/issues/256

### CODE REVIEW

File: src/mappings/SetController.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    for (const accountId of accountIds) {
+        const account = await Account.get(accountId);
+        if (!account) {
+            await new Account(accountId).save();
+        }
+    }
+}
+
+export async function handleSetController(extrinsic: SubstrateExtrinsic): Promise<void> {
+    console.log(extrinsic)
```
Comment (ironoa) [inline src/mappings/SetController.ts]: to remove the console.log

File: src/mappings/SetController.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    const {
+        extrinsic: {
+            signer: id,
+            args: args,
+        },
+    } = extrinsic;
+    var controller = args.toString();
```
Comment (ironoa) [inline src/mappings/SetController.ts]: "let" is better than "var"

File: schema.graphql [hunk 3]
```diff
@@ -85,3 +85,9 @@ type BatchParameter @entity {
   name: String!
   value: String!
 }
+
+type SetController @entity {
+  id: ID! #blockHeight-eventIdx
+  stash: String!
+  controller: String!
```
Comment (ironoa) [inline schema.graphql:92]: extrinsic: Extrinsic!

would be nice to have the extrinsic entity querable from that entity

Comment (Nexus2k) [reply schema.graphql:92]: I'm not sure what you mean. What's the advantage of that? I'm only interested in the stash-controller fields.

Comment (ironoa) [reply schema.graphql:92]: then let's leave it as it is

File: src/mappings/SetController.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+//https://github.com/subquery/tutorials-entity-relations/blob/main/src/mappings/AccountTransfers.ts
+import { SubstrateExtrinsic } from '@subql/types';
+import { Account } from '../types/models/Account';
+import { SetController } from '../types/models/SetController';
+
+async function ensureAccounts(accountIds: string[]): Promise<void> {
```
Comment (ironoa) [inline src/mappings/SetController.ts]: this can be removed

File: src/mappings/SetController.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+//https://github.com/subquery/tutorials-entity-relations/blob/main/src/mappings/AccountTransfers.ts
+import { SubstrateExtrinsic } from '@subql/types';
+import { Account } from '../types/models/Account';
```
Comment (ironoa) [inline src/mappings/SetController.ts]: this can be removed (Account)

File: src/mappings/SetController.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+export async function handleSetController(extrinsic: SubstrateExtrinsic): Promise<void> {
+    const {
+        extrinsic: {
+            signer: id,
+            args: args,
+        },
+    } = extrinsic;
+    let controller = args.toString();
```
Comment (ironoa) [inline src/mappings/SetController.ts:12]: this is far from the approach we took so far, did you do it on purpose ? i.e. sub entities with name-value

**Comment by ironoa** [comment]
FYI: this approach will make you loose all the bond and setController extrinsics nested in a batch call extrinsic, is that what you want ?

File: src/mappings/SetController.ts (hunk 6 above)
Comment (Nexus2k) [reply src/mappings/SetController.ts:12]: Yeah I figured out that the toString() call reliably just returned the value I'm interested in no matter the runtime version.

**Comment by Nexus2k** [comment]
> FYI: this approach will make you loose all the bond and setController extrinsics nested in a batch call extrinsic, is that what you want ?

I think that's ok for now.

**Pull request merged**